src/datepicker/datepicker.js

```diff
@@ -127,6 +127,16 @@ angular.module('ui.bootstrap.datepicker', ['ui.bootstrap.dateparser', 'ui.bootst
     return arrays;
   };
 
+  // Fix a hard-reprodusible bug with timezones
+  // The bug depends on OS, browser, current timezone and current date
+  // i.e.
+  // var date = new Date(2014, 0, 1);
+  // console.log(date.getFullYear(), date.getMonth(), date.getDate(), date.getHours());
+  // can result in "2013 11 31 23" because of the bug.
+  this.fixTimeZone = function(date) {
+    date.setHours(date.getHours() === 23 ? date.getHours() + 2 : 0);
```

Comment:
A micro-optimization here would be to do
<pre>
var hours = date.getHours();
date.setHours(hours === 23 ? hours + 2 : 0);
</pre>